### DISCUSSION TITLE
Binding not working on a custom control property

### DISCUSSION THREAD

**Discussion by Jeremy-Code-F** [AvaloniaUI/Avalonia#9039] [Q&A] [answered]
Hi,

I'm trying to implement a simple custom control that defines a custom property.

Custom Control backing code 
```
// TextboxTooltip.cs
using System;
using Avalonia;
using Avalonia.Controls;
using Avalonia.Markup.Xaml;
using BindingMre.ViewModels;
using ReactiveUI;

namespace BindingMre.Views.CustomControls;

public class TestViewModel: ViewModelBase
{
   private string _testProp = "Default Test Prop Value";

   public string TestProp
   {
      get => _testProp;
      set => this.RaiseAndSetIfChanged(ref this._testProp, value);
   }

   public ref String GetBackingProp()
   {
      return ref this._testProp;
   }
}
public partial class TextboxTooltip: UserControl
{
   private TestViewModel _vm = new TestViewModel();
   public TextboxTooltip()
   {
      this.DataContext = _vm; 
      this.InitializeComponent();
   }

   private void InitializeComponent()
   {
      AvaloniaXamlLoader.Load(this);
   }
   
   public string? TestProp
   {
      get => this._vm.TestProp;
      set => SetAndRaise(TestPropProperty, ref _vm.GetBackingProp(), value);
   }
   public static DirectProperty<TextboxTooltip, string?> TestPropProperty=
      AvaloniaProperty.RegisterDirect<TextboxTooltip, string?>(nameof(TestProp), o => o.TestProp, (o,v) => o.TestProp = v);
   
}
```

Custom Control view

```
// TextboxTooltip.axaml
<StackPanel xmlns="https://github.com/avaloniaui"
          xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
          xmlns:vm="clr-namespace:BindingMre.Views.CustomControls"
          x:Class="BindingMre.Views.CustomControls.TextboxTooltip"
          x:CompileBindings="True"
          >
    <Design.DataContext>
       <vm:TestViewModel></vm:TestViewModel>>
    </Design.DataContext>
    <TextBlock Text="Display Text"></TextBlock>
    <ToolTip.Tip>
        <TextBlock x:DataType="vm:TestViewModel" Text="{Binding TestProp}"></TextBlock>
    </ToolTip.Tip>
</StackPanel>

```

Now when I go to use it, the ```TestProp``` works when I assign it a String directly, but when I bind it to a string in MainWindowViewModel it no longer works

```
<StackPanel Orientation="Vertical">
        <cc:TextboxTooltip TestProp="Something here"></cc:TextboxTooltip>
        <cc:TextboxTooltip TestProp="{Binding Greeting}"></cc:TextboxTooltip>
    </StackPanel>
```
![image](https://user-images.githubusercontent.com/45114285/192659419-cecf843b-2d60-417c-8255-c3187b00c4a1.png)
![image](https://user-images.githubusercontent.com/45114285/192659438-013bc362-a1b5-4ca0-9fe0-37db079aa546.png)

Is there something I'm missing in order to get this working? 

I've setup an Mre [here](https://bitbucket.org/jeremycodef/bindingmre/src/master/BindingMre/) to show the exact issue I'm seeing.

Thanks!

**Comment by timunie** [comment, marked as answer]
Hi @Jeremy-Code-F , 

`UserControl` is more like a "Page" or a "Container" to display the data of the ViewModel. What you are trying to do is a custom control. In that case derive from `TemplatedControl` and provide a `Style` for this control.

See: https://docs.avaloniaui.net/docs/authoring-controls/types-of-control#templated-controls  

You can look up in third party libs how they defined own controls. A collection can be found here: https://github.com/AvaloniaCommunity/awesome-avalonia#controls 

Happy coding
Tim

**Reply by Jeremy-Code-F** [reply]
Tyvm!! Managed to get something working after deriving from ```TemplatedControl``` instead of ```UserControl.```
